flip/uploader/uploader.go

```diff
@@ -95,28 +99,49 @@ func New(cl *client.Client, factoryID string) (*Uploader, error) {
 // UploadSession uploads files based on the given session. If the upload has
 // failed or has been stopped, it is possible to resume it using this method.
 // Uploading stops on the first worker error and returns it.
-func (u *Uploader) UploadSession(s *Session, r io.ReaderAt, size int64) error {
-	return u.upload(s, r, size)
+func (u *Uploader) UploadSession(s *Session, r io.ReaderAt, size int64, extra_files *ExtraFileInfo) error {
```

Comment:
if ExtraFileInfo type wasn't a slice by itself: 

```
type ExtraFileInfo struct {
 	Tag   string
 	Files []string
 }
```

And you would turn signature into this:

`func (u *Uploader) UploadSession(s *Session, r io.ReaderAt, size int64, extraFiles ...ExtraFileInfo)`

then you would maintain backward compatibility, because it would be optional